sim/xcelium/__init__.py

```diff
@@ -0,0 +1,394 @@
+      f.write(f'{self.generate_saif_tcl_cmd()}\n')
+    
+    # Execute
+    f.write("run \n")
+    
+    # Close databases and dumps properly.
+    f.write("dumpsaif -end \n")
+    f.write("database -close *db \n")
+    f.write("exit")
+    f.close()  
+    return True
+
+  def compile_xrun(self) -> bool:
+    
+    if not os.path.isfile(self.xcelium_bin):
+      self.logger.error(f"Xcelium (xrun) binary not found at {self.xcelium_bin}.")
+      return False
+  
+    if not self.check_input_files(self.xcelium_ext):
+      return False
+
+    # Gather complation-only options
+    compile_opts  = self.get_setting(f"{self.tool_config_prefix}.compile_opts", [])       
+    compile_opts.append("-logfile xrun_compile.log")
+    compile_opts = ('COMPILE', compile_opts)
+    
+    arg_file_path = self.generate_arg_file("xrun_compile.arg", "HAMMER-GEN XRUN COMPILE ARG FILE", [compile_opts])
+    args =[self.xcelium_bin]
+    args.append(f"-compile -f {arg_file_path}")
+    
+    self.run_executable(args, cwd=self.run_dir)
+    HammerVLSILogging.enable_colour = True
+    HammerVLSILogging.enable_tag = True
+    return True
+    
+  def elaborate_xrun(self) -> bool: 
+    sim_opts  = self.extract_sim_opts()[1]
+    elab_opts = self.get_setting(f"{self.tool_config_prefix}.elab_opts", [])
+    elab_opts.append("-logfile xrun_elab.log")
+    elab_opts.append("-glsperf")
+    elab_opts.append("-genafile access.txt")  
+    
+    if self.level.is_gatelevel():
+      elab_opts.extend(self.get_verilog_models())    
+      if sim_opts["timing_annotated"]:
+        self.generate_sdf_cmd_file()
+        elab_opts.append(f"-sdf_cmd_file {self.sdf_cmd_file}")  
+        elab_opts.append("-sdf_verbose")
+        elab_opts.append("-negdelay")
+      else:
+        elab_opts.append("-notimingchecks")
+        elab_opts.append("-delay_mode zero")
+    else:
+      elab_opts.append("-notimingchecks")
+      elab_opts.append("-delay_mode zero")
+    elab_opts = ('ELABORATION', elab_opts)
+    
+    arg_file_path = self.generate_arg_file("xrun_elab.arg", "HAMMER-GEN XRUN ELAB ARG FILE", [elab_opts])
+    args =[self.xcelium_bin]
+    args.append(f"-elaborate -f {arg_file_path}")
+    
+    self.run_executable(args, cwd=self.run_dir)
+    return True
+
+  def sim_xrun(self) -> bool:
+    
+    sim_opts = self.get_setting(f"{self.sim_input_prefix}.options", [])
+    sim_opts_removal  = ["tb_name", "input_files", "incdir"]
+    xrun_opts_removal = ["enhanced_recompile"]
+    sim_opts = ('SIMULATION', sim_opts)
+    
+    arg_file_path = self.generate_arg_file("xrun_sim.arg", "HAMMER-GEN XRUN SIM ARG FILE", [sim_opts],
+                                           sim_opt_removal = sim_opts_removal,
+                                           xrun_opt_removal = xrun_opts_removal)    
+    args =[self.xcelium_bin]
+    args.append(f"-R -f {arg_file_path} -input {self.sim_tcl_file}")
+
+    self.generate_sim_tcl() 
+    self.run_executable(args, cwd=self.run_dir)
```

Comment:
This should be conditional on `sim.inputs.execute_sim`

Reply:
Fixed!